Py_FS/wrapper/nature_inspired/SSA.py

```diff
@@ -0,0 +1,180 @@
+"Jain, M., Singh, V., & Rani, A. (2019). A novel nature-inspired algorithm for optimization: Squirrel search algorithm.
+Swarm and evolutionary computation, 44, 148-175."
+
+"""
+
+import math
+
+import numpy as np
+from sklearn import datasets
+
+from Py_FS.wrapper.nature_inspired._utilities_test import initialize as initialize_
+from Py_FS.wrapper.nature_inspired._utilities_test import sort_agents as sort_agents_
+from Py_FS.wrapper.nature_inspired.algorithm import Algorithm
+
+
+class SSA(Algorithm):
+
+    # Squirrel Search Algorithm
+    ############################### Parameters ####################################
+    #                                                                             #
+    #   num_agents: number of squirrels                                           #
+    #   max_iter: maximum number of generations                                   #
+    #   train_data: training samples of data                                      #
+    #   train_label: class labels for the training samples                        #
+    #   obj_function: the function to maximize while doing feature selection      #
+    #   trans_function_shape: shape of the transfer function used                 #
+    #   save_conv_graph: boolean value for saving convergence graph               #
+    #                                                                             #
+    ###############################################################################
+
+    def __init__(self,
+                 num_agents,
+                 max_iter,
+                 train_data,
+                 train_label,
+                 save_conv_graph=False,
+                 trans_func_shape='s',
+                 seed=0):
+
+        super().__init__(num_agents=num_agents,
+                         max_iter=max_iter,
+                         train_data=train_data,
+                         train_label=train_label,
+                         save_conv_graph=save_conv_graph,
+                         trans_func_shape=trans_func_shape,
+                         seed=seed)
+
+        self.algo_name = 'SSA'
+        self.agent_name = 'Squirrel'
+        self.algo_params = {}
+
+    def user_input(self):
+        # initializing parameters
+        pdp = self.algo_params['pdp'] = 0.1
+        row = self.algo_params['row'] = 1.204
+        V = self.algo_params['V'] = 5.25
+        S = self.algo_params['S'] = 0.0154
+        cd = self.algo_params['cd'] = 0.6
+        CL = self.algo_params['CL'] = 0.7
+        hg = self.algo_params['hg'] = 1
+        sf = self.algo_params['sf'] = 18
+        Gc = self.algo_params['Gc'] = 1.9
+        D1 = self.algo_params['D1'] = 1 / ((2 * row * V) ** (2 * S * cd))
+        L = self.algo_params['L'] = 1 / ((2 * row * V) ** (2 * S * CL))
+        tanpi = self.algo_params['tanpi'] = D1 / L
+        dg = self.algo_params['dg'] = hg / (tanpi * sf)
+

```

Comment:
Allow the user to enter the values of the parameters [Refer to the user_input() method of WOA_test.py].
Initialize the other variables(D1, L, tanpi, dg) in the initialize() method.

Reply:
Done.

Reply:
Why are you taking these local variables pdp, row, V, etc. ? These are not being used.
self.algo_params['pdp'], self.algo_params['row'], etc. should suffice.